xarray/core/dataarray.py

```diff
@@ -1195,6 +1195,55 @@ def sel(
         Dataset.sel
         DataArray.isel
 
+        Examples
+        --------
+        >>> ds = xr.tutorial.open_dataset("air_temperature")
```

Comment:
Thanks for the tip. I've implemented your suggestion in `DataArray.sel`. For `DataArray.isel` `method='nearest` doesn't appear to be an option? So I tried this, but it doesn't seem to work. I'm not sure why? 
``` python
da = xr.DataArray(
                np.arange(25).reshape(5, 5),
                coords={"x": np.arange(5), "y": np.arange(5)},
                dims=("x", "y")
            )
tgt_x = xr.DataArray(np.linspace(0, 2, num=3), dims="points")
tgt_y = xr.DataArray(np.linspace(0, 2, num=3), dims="points")

da.isel(x=tgt_x, y=tgt_y) 
```

Reply:
`isel` expects `int`, but `linspace` returns `float`. Since you're using it to generate indices, I would use `arange` instead.

Reply:
Working on building an example for `Dataset.sel` using `method='nearest'`

```python
import numpy as np
import pandas as pd
import xarray as xr

ds = xr.Dataset(
	data_vars=dict(
			temperature=(["x", "y", "time"], np.arange(125).reshape(5, 5, 5))
		),
		coords=dict(
			lon=(["x", "y"], np.arange(25).reshape(5, 5)),
			lat=(["x", "y"], np.arange(25).reshape(5, 5)),
			time=pd.date_range("2014-09-06", periods=5),
		),
		attrs=dict(description="Weather related data.")
)

# Define target latitude and longitude
tgt_x = xr.DataArray(np.arange(0, 5), dims="points")
tgt_y = xr.DataArray(np.arange(0, 5), dims="points")

# Dataset.sel
ds_sel = ds.sel(x=tgt_x, y=tgt_y, method='nearest')
```

However I get the following error, which I don't understand. I assume my coordinate definitions are inconsistent? 
```
Traceback (most recent call last):
  File "/Users/erke2265/Documents/xarray/xarray/core/indexing.py", line 257, in remap_label_indexers
    index = data_obj.indexes[dim]
  File "/Users/erke2265/Documents/xarray/xarray/core/indexes.py", line 64, in __getitem__
    return self._indexes[key]
KeyError: 'x'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tmp.py", line 24, in <module>
    ds_sel = ds.sel(x=tgt_x, y=tgt_y, method='nearest')
  File "/Users/erke2265/Documents/xarray/xarray/core/dataset.py", line 2275, in sel
    self, indexers=indexers, method=method, tolerance=tolerance
  File "/Users/erke2265/Documents/xarray/xarray/core/coordinates.py", line 398, in remap_label_indexers
    obj, v_indexers, method=method, tolerance=tolerance
  File "/Users/erke2265/Documents/xarray/xarray/core/indexing.py", line 262, in remap_label_indexers
    "cannot supply ``method`` or ``tolerance`` "
ValueError: cannot supply ``method`` or ``tolerance`` when the indexed dimension does not have an associated coordinate.
```

Reply:
the reason is that you define `lat` and `lon` as 2D non-dimension coordinates and `x` and `y` as dimensions without coordinates. As the error says, `sel` with `method` or `tolerance` only works with dimensions with coordinates, so you need to add coordinates to `x` and `y` (actually, I would have expected `sel` to fail on all dimensions without coordinates, so I'm a bit surprised the error message implies that could work).

This also means that you don't have to define `lat` and `lon` since you won't be using them in this example (it's fine to keep it if you want to use them in later examples, though).